doc/flame/inputs/gesture-input.md

```diff
@@ -282,6 +282,25 @@ class MyGame extends FlameGame with HasTappables {
 **Note**: `HasTappables` uses an advanced gesture detector under the hood and as explained
 further up on this page it shouldn't be used alongside basic detectors.
 
+To recognize whether a `Tappable` added to the game handled an event, the `boolean` returned by
+`super` can be checked.
+
+In the following example it can be seen how it is used with `onTapDown`, the same technique can also
+be applied to `onTapUp` and `onTapCancel`.
+
+```dart
+class MyGame extends FlameGame with HasTappables {
+  @override
+  bool onTapDown(int pointerId, TapDownInfo info) {
+    final handled = !super.onTapDown(info);
+    if(handled) {
+      // Do something if a child handled the event
+    }
+    return handled;
+  }
```

Comment:
Compare with the following API:
```dart
void onTapDown(TapDownEvent event) {
  super.onTapDown(event);
  if (event.handled) {
    // do something
  }
}
```

Reply:
I agree that is nicer, I don't think we wanted mutable events though (ping @ErickZanardo). But I think it would be good to go that way anyways, because then this wouldn't be a breaking change.

Reply:
We have plenty of mutable vectors everywhere, not sure why mutable events are a problem then...

Reply:
That's not the same thing though, because you always change and get the `Vector2` from a known object, here "something unknown" further up in the chain will modify the value. But as I said, I think it's still worth it.

Reply:
Yeah, not a big fan of mutating events, I mean a event should be a snapshot of something that have already happened, and you can't change the past (at least AFAIK :P)

But I wonder if we couldn't think on an alternative to having to check if the event was already handled in other to do our logic, maybe the engine could check that and only call the method if it wasn't handled yet?

Reply:
Possibly, but that closes down some options for the users, like handling events differently depending on whether it was handled or not, and it would still be a breaking change. The `Has*` game mixins should have a bit more control than the component mixins I think.

Having a mutable `handled` flag on the event isn't really changing the past imho, since we know the flow of these events. If the user isn't passing the event somewhere else, and then reusing it later, but that would be a really strange design... And if they look at that event later for some weird reason and the flag has changed, then it is actually true that the event was handled.

Isn't it better with a flag than a breaking change at least?

Reply:
What if we had a method instead of a flag then? Like a `stopPropagation` method, so internally we could still make the event be immutable and we would have the same scenario of the attribute.

Reply:
This is not for the propagation stopping or not, this is for knowing whether an event was handled or not in the `Has*` methods.

Reply:
I think what is confusing me, is all that if you did on the return value of the parent method right? But looking deeper now I bet you did that because with a return type of bool we had to use must call super....

If we had the mutating flag, we wouldn't need to force the user to check that right? Because that is my biggest concern, forcing the user to always write that, which will pollute the code for the most cases.

If this assumption is correct, I think I can live with the mutating variable as its benefits will overgrow

Reply:
I'll change the PR and then we can compare the commits to see which way we like better? :)

Reply:
Sounds good!